unpack_action.go

```diff
@@ -51,7 +56,23 @@ func (pf *UnpackAction) Verify(context *DebosContext) error {
 
 func (pf *UnpackAction) Run(context *DebosContext) error {
 	pf.LogStart()
-	infile := path.Join(context.artifactdir, pf.File)
+	var origin string
+
+	if len(pf.Origin) > 0 {
+		//Trying to get a filename from origins first
+		val, found := context.origins[pf.Origin]
+		if !found {
+			return fmt.Errorf("Origin not found '%s'", pf.Origin)
+		}
+		origin = val
+	} else {
+		origin = context.artifactdir
+	}
+
+	infile, err := RestrictedPath(origin, pf.File)
```

Comment:
This probably means that function is another candidate to move to a utility file

Reply:
Yes. During refactoring.